Code/Tools/AssetProcessor/native/ui/ProcessesModel.cpp

```diff
@@ -0,0 +1,165 @@
+    {
+        auto itr = AZStd::find_if(
+            m_builders.begin(),
+            m_builders.end(),
+            [builderId](auto& obj)
+            {
+                return AZStd::get<0>(obj) == builderId;
+            });
+
+        if (itr != m_builders.end())
+        {
+            auto rowNumber = AZStd::distance(m_builders.begin(), itr);
+            Q_EMIT beginRemoveRows(QModelIndex(), rowNumber, rowNumber);
+            m_builders.erase(itr);
+            Q_EMIT endRemoveRows();
+        }
+    }
+
+    void ProcessesModel::DoUpdate(AZ::Uuid builderId, bool process, bool connection, bool busy, QString file)
+    {
+        auto itr = AZStd::find_if(
+            m_builders.begin(),
+            m_builders.end(),
+            [builderId](auto& obj)
+            {
+                return AZStd::get<0>(obj) == builderId;
+            });
+
+        if (itr != m_builders.end())
+        {
+            auto rowNumber = AZStd::distance(m_builders.begin(), itr);
+            auto& [builderUuid, builderProcess, builderConnection, builderBusy, builderFile] = *itr;
+
+            builderProcess = process;
+            builderConnection = connection;
+            builderBusy = busy;
+            builderFile = file;
+
+            Q_EMIT dataChanged(index(rowNumber, 0), index(rowNumber, 0));
+        }
+
+        const int numberBusyBuilders = AZStd::count_if(
+            m_builders.begin(),
+            m_builders.end(),
+            [](auto& obj)
+            {
+                return AZStd::get<3>(obj);
+            });
+
+        Q_EMIT UtilizationUpdate(m_builders.size(), numberBusyBuilders);
+    }
+
+    void ProcessesModel::OnStatusUpdate(AZ::Uuid builderId, bool process, bool connection, bool busy, QString file)
+    {
+        if (m_debounceTimer.isActive())
+        {
+            m_pendingUpdates[builderId] = { builderId, process, connection, busy, file };
+            return;
+        }
+
+        DoUpdate(builderId, process, connection, busy, file);
+        m_debounceTimer.start(200);
+    }
+
+    int ProcessesModel::rowCount(const QModelIndex& index) const
+    {
+        return index.isValid() ? 0 : m_builders.size();
+    }
+
+    QVariant ProcessesModel::data(const QModelIndex& index, int role) const
+    {
+        if (!index.isValid())
+        {
+            return QVariant();
+        }
+
+        if (index.row() >= m_builders.size())
+        {
+            return QVariant();
+        }
+
+        auto& [uuid, process, connection, busy, file] = m_builders.at(index.row());
+
+        if (role == Qt::DisplayRole)
+        {
+            QString status = busy ? (m_showFilename ? file : "Busy") : connection ? "Idle" : "Boot";
+
+            return QStringLiteral("#%1 %2").arg(index.row(), 2, 10, QLatin1Char('0')).arg(status);
+        }
+
+        if (role == Qt::BackgroundRole)
+        {
+            if (busy)
+            {
+                return QColor(255, 236, 31);
```

Comment:
You can probably define these in a Qt file, maybe here? https://github.com/o3de/o3de/blob/development/Code/Tools/AssetProcessor/native/ui/style/AssetProcessor.qss

It will be easier to find and manage these colors if they're in a Qt settings file

Reply:
I asked about that and was told it would probably require writing a custom delegate to do a lookup in the stylesheet since the listview items aren't real widgets.